server/runtime_status.go

```diff
@@ -48,9 +48,14 @@ func (s *Server) Status(ctx context.Context, req *types.StatusRequest) (*types.S
 			}
 		}
 
-		// TODO: enable when CRI-O implemented Recursive Read-only (RRO) mounts
-		// rro := runtime.RuntimeSupportsMountFlag("rro")
 		rro := false
+		if runtime.RuntimeSupportsMountFlag("rro") {
+			// A given runtime might support Recursive Read-only (RRO) mounts,
+			// but the underlying kernel might not.
+			if err := checkKernelRROMountSupport(); err == nil {
```

Comment:
we call this a lot, potentially unnecessarily. I think we should check this in `initializeRuntimeFeatures()` in `pkg/config/config.go` and if the kernel doesn't support the flag, then remove it from the features structure. Then, from that point on, we can just ask the runtime whether it's supported or not

Reply:
I see what you mean. I think we can plumb this like that. Let me see. :+1:

Reply:
We unmarshal the JSON atop the following:

_(from [opencontainers/runtime-spec/specs-go/features/features.go](https://github.com/opencontainers/runtime-spec/blob/main/specs-go/features/features.go#L5-L33)):_

```go
type Features struct {
	// OCIVersionMin is the minimum OCI Runtime Spec version recognized by the runtime, e.g., "1.0.0".
	OCIVersionMin string `json:"ociVersionMin,omitempty"`

	// OCIVersionMax is the maximum OCI Runtime Spec version recognized by the runtime, e.g., "1.0.2-dev".
	OCIVersionMax string `json:"ociVersionMax,omitempty"`

	// Hooks is the list of the recognized hook names, e.g., "createRuntime".
	// Nil value means "unknown", not "no support for any hook".
	Hooks []string `json:"hooks,omitempty"`

	// MountOptions is the list of the recognized mount options, e.g., "ro".
	// Nil value means "unknown", not "no support for any mount option".
	// This list does not contain filesystem-specific options passed to mount(2) syscall as (const void *).
	MountOptions []string `json:"mountOptions,omitempty"`

	// Linux is specific to Linux.
	Linux *Linux `json:"linux,omitempty"`

	// Annotations contains implementation-specific annotation strings,
	// such as the implementation version, and third-party extensions.
	Annotations map[string]string `json:"annotations,omitempty"`

	// PotentiallyUnsafeConfigAnnotations the list of the potential unsafe annotations
	// that may appear in `config.json`.
	//
	// A value that ends with "." is interpreted as a prefix of annotations.
	PotentiallyUnsafeConfigAnnotations []string `json:"potentiallyUnsafeConfigAnnotations,omitempty"`
}
```

This is not the same as the CRI API type, which I think you had in mind here?